R/jaspHistogram.R

```diff
@@ -0,0 +1,229 @@
+#' @title Histograms and Density plots for JASP
+#'
+#' @description A plot histogram with three components.
+#' (1) A histogram, which can be tweaks with the same settings as \code{\link[graphics]{hist}}.
+#' (2) A density line, which can optionally be shaded.
+#' (3) rugs underneath the figure.
+#' Each of these components can be enabled (or disabled) individually.
+#' Colors are taken from \code{graphOptions("palette")}.
+#'
+#' @param x, numeric, the data to show a histogram for
+#' @param xName, string, the title on the x-axis
+#' @param groupingVariable, factor, show histograms and/ or densities for individual groups?
+#' @param groupingVariableName, character, legend name of the grouping variable.
+#' @param histogram, logical, should a histogram be shown?
+#' @param histogramPosition, character, see argument \code{position} in \code{\link[ggplot2]{geom_histogram}}.
+#' @param binWidthType, string, type of bindwidth, matches the string values that can be passed to \code{\link[graphics]{hist}}.
+#' @param numberOfBins, if binWidthType is set to "manual", this value determines the number of bins.
+#' @param rugs, logical, should rugs be shown on the x-axis?
+#' @param rugsColor, logical, should the rugs be colored?
+#' @param density, logical, should a density be superimposed on the plot?
+#' @param densityColor, logical, should the density be colored?
+#' @param densityShade, logical, should the area underneath the density be shaded?
+#' @param densityShadeAlpha, numeric in \[0, 1\], transparancy for the shaded density.
+#' @param densityLineWidth, positive number, the line width of the superimposed density.
+#' @param hideYAxisLabels, logical, should the y-axis line be hidden? Defaults to \code{showDensity}.
+#' @example inst/examples/ex-jaspHistogram.R
+#' @export
+jaspHistogram <- function(
+  x, xName,
+  groupingVariable = NULL,
+  groupingVariableName = NULL,
+  histogram = TRUE,
+  histogramPosition = "dodge",
+  binWidthType = c("doane", "fd", "scott", "sturges", "manual"),
+  numberOfBins = NA,
+  rugs = FALSE,
+  rugsColor = FALSE,
+  density = FALSE,
+  densityColor = FALSE,
+  densityShade = FALSE,
+  densityShadeAlpha = 0.6,
+  densityLineWidth = 1,
+  hideYAxisLabels = density) {
+
+  # validate input
+  if (!is.vector(x, mode = "numeric"))
+    stop2("`x` must be a numeric vector but has class ")
+
+  if (missing(xName))
+    xName <- deparse1(substitute(x)) # identical to plot.default
```

Comment:
since you do this for `xName` (nice!), you could also do this for `groupingVariableName`.